bugrap/src/main/java/org/vaadin/bugrap/ui/ReportsModel.java

```diff
@@ -35,18 +47,44 @@ public void fillProjects(ComboBox<Project> combo) {
 		combo.setItems(projects);
 		combo.setSelectedItem(projects.get(0));
 	}
+	
+	public void fillPriorties(ComboBox<Priority> combo) {
+		combo.setItems(Priority.values());
+	}
+	
+	public void fillTypes(ComboBox<Type> combo) {
+		combo.setItems(Type.values());
+		combo.setItemCaptionGenerator(type -> StringUtil.converToCamelCaseString(type.toString()));
+	}
+	
+	public void fillStatuses(ComboBox<Status> combo) {
+		combo.setItems(Status.values());
+	}
 
-	public void processProjectChange(Optional<Project> selectedItem, ComboBox<ProjectVersion> cmbVersion) {
+	public void fillAssigned(ComboBox<Reporter> combo) {
+//		List<Reporter> reporters = new ArrayList<Reporter>();
+//		reporters.addAll();
+//		Collections.sort(reporters);
+		combo.setItems(getRepository().findReporters());
+	}
+	
+	public void processProjectChange(Optional<Project> selectedItem, ComboBox<ProjectVersion> cmbVersion, ComboBox<ProjectVersion> cmbVersionEdit) {
 		selectedProject = selectedItem.get();
 		List<ProjectVersion> projectVersions = new ArrayList<ProjectVersion>();
 		projectVersions.addAll(getRepository().findProjectVersions(selectedProject));
 		Collections.sort(projectVersions);
 		
 		cmbVersion.setItems(projectVersions);
+		cmbVersionEdit.setItems(projectVersions);
```

Comment:
Seems that the model is too coupled with the View. Maybe it would be better if it just returned the data needed by the View, which then sets itself accordingly.